### ISSUE TITLE
Failure: Cannot execute az acr build --registry <> --image <> .

### ISSUE THREAD

**Issue by armandasalmd** [Azure/azure-cli#18796]
## Describe the bug
I am trying to build my docker image and push it to azure registry

I was following this guide: https://circleci.com/blog/deploying-container-based-node-apps-to-azure/

**Command Name**
`az acr build`

**Logs this first:**
![image](https://user-images.githubusercontent.com/17089888/125057889-9dfea580-e0a1-11eb-8a0f-95ea3e01915f.png)

**Then errors:**

```
The command failed with an unexpected error. Here is the traceback:
'NoneType' object is not callable
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 231, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 657, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 691, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 328, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acr/build.py", line 106, in acr_build
TypeError: 'NoneType' object is not callable

```

## To Reproduce:

- `az acr build --registry {} --image {} {}`

## Expected Behavior

- build and push image to acr registry

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.8.9
Installer: MSI

azure-cli 2.26.0

```

**Comment by yonzhan** [comment]
route to service team

**Comment by jchomarat** [comment]
Apparently if we do not provide a platform_variant, it does not work any more (it worked in 2.25.0)

**Comment by codycodes** [comment]
I'm sure this will be resolved soon, but if you need to workaround the issue here are the steps I took:

If you're on Windows, you can most likely just install the earlier [release](https://github.com/Azure/azure-cli/releases)

For those like me who are using WSL or other *nix-based OS and are using `apt` as your package manager, here are the steps I took to downgrade:

```
# Use apt-cache to install different versions of a package

sudo apt-get update

# get the available versions using: apt-cache showpkg <package-name>
# install a specific version using: sudo apt-get install package=version -V

sudo apt-get install azure-cli=2.23.0-1~focal -V
```

More info: [How to Install Specific Version of Package using apt-get (linoxide.com)](https://linoxide.com/install-specific-version-package-apt-get/)

**Comment by northtyphoon** [comment]
The issue has been fixed since 2.26.1
